elasticsearch/src/main/scala/uk/ac/wellcome/elasticsearch/IndexConfig.scala

```diff
@@ -22,14 +23,25 @@ object RefreshInterval {
   final case class On(duration: FiniteDuration) extends RefreshInterval
 }
 
-trait IndexConfig {
-  def mapping: MappingDefinition
-  def analysis: Analysis
-  def shards: Int = 1
-  def refreshInterval: RefreshInterval = RefreshInterval.Default
-}
+case class IndexConfig(mapping: MappingDefinition,
+                       analysis: Analysis,
+                       shards: Int = 1,
+                       refreshInterval: RefreshInterval =
+                         RefreshInterval.Default)
+
+object IndexConfig {
+  def empty =
+    IndexConfig(
+      analysis = Analysis(analyzers = List()),
+      mapping = MappingDefinition())
 
-object NoStrictMapping extends IndexConfig {
-  val analysis: Analysis = Analysis(analyzers = List())
-  val mapping: MappingDefinition = MappingDefinition()
+  implicit class IndexConfigOps(indexConfig: IndexConfig) {
+    def withRefreshIntervalFromConfig(config: Config): IndexConfig = {
+      val isReindexing = config.getBoolean(s"es.is_reindexing")
```

Comment:
```suggestion
      val isReindexing = config.getBoolean("es.is_reindexing")
```

Wherever you put this, you don't need s-string interpolation.